src/future/shared.rs

```diff
@@ -288,6 +288,7 @@ impl<E> fmt::Display for SharedError<E>
 impl<E> error::Error for SharedError<E>
     where E: error::Error,
 {
+    #[allow(deprecated)]
     fn description(&self) -> &str {
```

Comment:
`Error::description` is a required method in rust 1.15: https://doc.rust-lang.org/1.15.0/std/error/trait.Error.html#tymethod.description